### PR TITLE
refactor(state): refine naming and conventions in state management layer

### PR DESCRIPTION
**Pull request by pannpers**
## Description

Refine the state management layer naming and conventions for consistency and readability.

- **OnboardingStep**: numeric values → descriptive string values (`'lp'`, `'discovery'`, `'dashboard'`, `'detail'`, `'my-artists'`, `'completed'`)
- **State key**: `guestArtists` → `guest` to match `guest/` action namespace
- **Terminology**: all `tutorial` references → `onboarding` (route data, getters, methods, aria-labels, i18n)
- **Logger middleware**: `console.log` → Aurelia `ILogger` via factory pattern, dev-only
- **Dead code**: remove unused `LOADING` and `SIGNUP` steps, no backward compat
- **Step ordering**: `STEP_ORDER` array + `stepIndex()` replaces numeric `>=` comparison
- **Onboarding detection**: `ONBOARDING_STEPS` Set replaces numeric range check
- **Tests**: comprehensive edge-case unit tests for all reducer transitions (19 reducer + 12 middleware tests)

## Type of Change

- [ ] feat: A new feature
- [ ] fix: A bug fix
- [ ] docs: Documentation only changes
- [x] refactor: A code change that neither fixes a bug nor adds a feature
- [ ] test: Adding missing tests or correcting existing tests
- [ ] build: Changes that affect the build system or external dependencies
- [ ] chore: Other changes that don't modify src or test files

### Automated Tests
- [x] `npm run lint` passed
- [x] `npm run test` passed (455 tests)
- [x] `npm run build` passed

### Manual Verification
- Verified all `guestArtists` references removed from src/ and test/
- Verified all `tutorial` references removed from src/ and test/
- Verified OnboardingStep has exactly 6 string entries, no LOADING/SIGNUP
- Verified logging middleware only registered in dev mode

## Checklist
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings

close: #207

### CODE REVIEW

File: test/services/onboarding-service.spec.ts [hunk 1]
```diff
@@ -1,17 +1,30 @@
...
 import { createMockLogger } from '../helpers/mock-logger'
+import { createMockStore } from '../helpers/mock-store'
 
-// Import the real service (not mocked) for integration-style tests
 const { OnboardingService, OnboardingStep } = await import(
 	'../../src/services/onboarding-service'
 )
 
-function createService(): InstanceType<typeof OnboardingService> {
+function createService(overrides: { step?: number } = {}) {
```
[the remark this answers is not in the document]
Comment (pannpers) [reply test/services/onboarding-service.spec.ts]: ✅ Addressed in 9ebbdcf.

**What was changed**: Changed `step?: number` to `step?: OnboardingStepValue` in the test helper, using a local type alias derived from the dynamically imported `OnboardingStep`.
**Why**: The type annotation was stale — OnboardingStep values are now string literals, not numbers.
